core/commands/files.go

```diff
@@ -1063,66 +971,48 @@ Remove files or directories.
 		dir, name := gopath.Split(path)
 		parent, err := mfs.Lookup(nd.FilesRoot, dir)
 		if err != nil {
-			res.SetError(fmt.Errorf("parent lookup: %s", err), cmdkit.ErrNormal)
-			return
+			return fmt.Errorf("parent lookup: %s", err)
 		}
 
 		pdir, ok := parent.(*mfs.Directory)
 		if !ok {
-			res.SetError(fmt.Errorf("no such file or directory: %s", path), cmdkit.ErrNormal)
-			return
+			return fmt.Errorf("no such file or directory: %s", path)
 		}
 
-		var success bool
-		defer func() {
-			if success {
-				err := pdir.Flush()
-				if err != nil {
-					res.SetError(err, cmdkit.ErrNormal)
-					return
-				}
-			}
-		}()
```

Comment:
:+1: for the removal of the defer, it isn't really needed and the logic is simpler now.